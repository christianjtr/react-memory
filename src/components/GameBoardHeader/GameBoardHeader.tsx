import React from 'react';

export interface GameBoardHeaderProps {
  title?: string;
}

const GameBoardHeader: React.FC<GameBoardHeaderProps> = (
  props: GameBoardHeaderProps,
): React.ReactElement => {
  const { title = 'Github Memory' } = props;

  return <h1 className="my-5 text-3xl text-center">{title}</h1>;
};

export default GameBoardHeader;
