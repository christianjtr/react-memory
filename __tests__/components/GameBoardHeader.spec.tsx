import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, cleanup } from '@testing-library/react';
import { createRenderer } from 'react-test-renderer/shallow';

import GameBoardHeader from '../../src/components/GameBoardHeader/GameBoardHeader';

describe('GameBoardHeader component', () => {
  afterEach(() => {
    cleanup();
  });

  it('should render the component correctly', () => {
    const renderer = createRenderer();
    const component = renderer.render(<GameBoardHeader />);

    expect(component).toBeDefined();
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it('should have the default title', () => {
    render(<GameBoardHeader />);

    const defaultVale = 'Github Memory';

    const h1Element = screen.getByText(defaultVale);

    expect(h1Element.textContent).toEqual(defaultVale);
  });

  it('should have a custom title', () => {
    const customTitle = 'Custom title';

    render(<GameBoardHeader title={customTitle} />);

    const h1Element = screen.getByText(customTitle);

    expect(h1Element.textContent).toEqual(customTitle);
  });
});
