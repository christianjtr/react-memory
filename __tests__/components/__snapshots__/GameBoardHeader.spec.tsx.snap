// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoardHeader component should render the component correctly 1`] = `
<h1
  className="my-5 text-3xl text-center"
>
  Github Memory
</h1>
`;
